### PR TITLE
scaleway: update to new sdk

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: provider/scaleway/scaleway_discover.go [hunk 1]
```diff
@@ -5,20 +5,31 @@ import (
...
 	return `Scaleway:
 
-    provider:     "scaleway"
-    organization: The Scaleway organization access key
-    tag_name:     The tag name to filter on
-    token:        The Scaleway API access token
-    region:       The Scalway region
+    provider:           "scaleway"
+    name:               The name of the instance to filter on
+    organization:       The Scaleway organization access key
```
Comment (kindermoumoute) [inline provider/scaleway/scaleway_discover.go:21]: document project_id

File: provider/scaleway/scaleway_discover.go [hunk 2]
```diff
@@ -28,54 +39,151 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
...
 
-	organization := args["organization"]
+	addrType := args["addr_type"]
 	tagName := args["tag_name"]
-	token := args["token"]
-	region := args["region"]
-
-	l.Printf("[INFO] discover-scaleway: Organization is %q", organization)
-	l.Printf("[INFO] discover-scaleway: Region is %q", region)
+	commercialType := args["commercial_type"]
```
Comment (kindermoumoute) [inline provider/scaleway/scaleway_discover.go:47]: nit: rename to server_type

File: provider/scaleway/scaleway_discover.go [hunk 3]
```diff
@@ -5,20 +5,31 @@ import (
...
-    tag_name:     The tag name to filter on
-    token:        The Scaleway API access token
-    region:       The Scalway region
+    provider:           "scaleway"
+    name:               The name of the instance to filter on
+    organization:       The Scaleway organization access key
+    tag_name:           The tag names to filter on (use "," as a separator)
+    commercial_type:    The commercial type to filter on
+    private_network_id: The private network id to filter on
+    token:              The Scaleway API secret key
```
Comment (kindermoumoute) [inline provider/scaleway/scaleway_discover.go:25]: secret-key?

File: provider/scaleway/scaleway_discover.go [hunk 4]
```diff
@@ -28,54 +39,151 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
...
+	zone := args["zone"]
+	if zone != "" {
+		z, err := scw.ParseZone(zone)
+		if err != nil {
+			return nil, fmt.Errorf("error during zone parsing: %s", err)
 		}
+		argsProfile.DefaultZone = scw.StringPtr(z.String())
+	}
+
+	organization := args["organization"]
```
Comment (kindermoumoute) [inline provider/scaleway/scaleway_discover.go:155]: can we rename to organization-id?

File: provider/scaleway/scaleway_discover.go [hunk 5]
```diff
@@ -28,54 +39,151 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
...
+	}
+	configProfile := &config.Profile
+
+	// Environment config loading
+	envProfile := scw.LoadEnvProfile()
+
+	// Command line argument loading
+	argsProfile := &scw.Profile{}
+
+	projectID := args["project-id"]
```
Comment (kindermoumoute) [inline provider/scaleway/scaleway_discover.go:136]: why do you use `-` sometimes and `_` other time?

File: provider/scaleway/scaleway_discover.go [hunk 6]
```diff
@@ -28,54 +39,151 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
...
+
+	profile := scw.MergeProfiles(defaultZoneProfile, configProfile, envProfile, argsProfile)
+
+	l.Printf("[INFO] discover-scaleway: Organization is %q", *profile.DefaultOrganizationID)
+	l.Printf("[INFO] discover-scaleway: Project is %q", *profile.DefaultProjectID)
+	l.Printf("[INFO] discover-scaleway: Secret is %q", hideSecretKey(*profile.SecretKey))
+	l.Printf("[INFO] discover-scaleway: Zone is %q", *profile.DefaultZone)
+
+	opts := []scw.ClientOption{
+		scw.WithUserAgent(fmt.Sprintf("go-discover")),
```
Comment (kindermoumoute) [inline provider/scaleway/scaleway_discover.go:168]: is there a version of go-discover we can add in user agent?

**Review by kindermoumoute — CHANGES_REQUESTED** [review]

File: provider/scaleway/scaleway_discover.go (hunk 6 above)
Comment (remyleone) [reply provider/scaleway/scaleway_discover.go]: Unfortunately not at the moment.

Comment (remyleone) [reply provider/scaleway/scaleway_discover.go]: @dnephin do you think it could be interesting to add support for a go-discover version?.

**Review by kindermoumoute — APPROVED** [review]
LGTM

**Comment by remyleone** [comment]
@dnephin what is the process for getting a contribution merged into this repository? Is there something we can do?

**Comment by rozhok** [comment]
Can we please merge this?

**Comment by remyleone** [comment]
@rboyer up :)

**Comment by alviss7** [comment]
@CalebAlbers Will we ever have the chance to have this PR accepted and merge? If there is something missing, can you at least tell us?

**Pull request closed without merging**